testing/trino-faulttolerant-tests/src/test/java/io/trino/faulttolerant/TestDistributedFaultTolerantEngineOnlyQueries.java

```diff
@@ -97,4 +103,48 @@ t2 AS (
 
         assertUpdate("DROP TABLE " + tableName);
     }
+
+    @Test(timeOut = 30000)
+    public void testMetadataOnlyQueries()
+            throws InterruptedException
+    {
+        // enforce single task uses whole node
+        Session highTaskMemorySession = Session.builder(getSession())
+                .setSystemProperty("fault_tolerant_execution_coordinator_task_memory", "500GB")
+                .setSystemProperty("fault_tolerant_execution_task_memory", "500GB")
+                .build();
+
+        ExecutorService backgroundExecutor = newCachedThreadPool();
+        try {
+            Future<?> longQueryFuture = backgroundExecutor.submit(() -> {
+                String longQuery = "select count(*) FROM lineitem l1 cross join lineitem l2 cross join lineitem l3 where l1.orderkey * l2.orderkey * l3.orderkey = 1";
+                query(highTaskMemorySession, longQuery);
+            });
+
+            Thread.sleep(5000); // wait for long query startup
+            assertThat(longQueryFuture).as("long query finished to early").isNotDone();
+
+            assertThat(query("DESCRIBE lineitem")).succeeds();
+            assertThat(query("SHOW TABLES")).succeeds();
+            assertThat(query("SHOW TABLES LIKE 'line%'")).succeeds();
+            assertThat(query("SHOW CATALOGS")).succeeds();
+            assertThat(query("SHOW CATALOGS LIKE 'def%'")).succeeds();
+            assertThat(query("SHOW FUNCTIONS")).succeeds();
+            assertThat(query("SHOW FUNCTIONS LIKE 'split%'")).succeeds();
+            assertThat(query("SHOW COLUMNS FROM lineitem")).succeeds();
+            assertThat(query("SHOW SESSION")).succeeds();
+
+            // check non-metadata queries still wait for resources
+            Future<?> nonMetadataQueryFuture = backgroundExecutor.submit(() -> {
+                query("select count(*) from nation");
+            });
+            Thread.sleep(2000);
```

Comment:
can we do without sleep (7s total)?

Reply:
Is there a way to test if query already started introspecting query runner?

Reply:
Removed most (just 1s left)